File: js/tests/filterStringsFromArray.test.js
const filterArray = require("../filterStringsFromArray");

const data = [
  [
    [1, 2, "a", "b"],
    [1, 2]
  ],
  [
    [1, "a", "b", 0, 15],
    [1, 0, 15]
  ],
  [
    [1, 2, "aasf", "1", "123", 123],
    [1, 2, 123]
  ],
  [
    ["jsyt", 4, "yt", 6],
    [4, 6]
  ],
  [
    ["r", 5, "y", "e", 8, 9],
    [5, 8, 9]
  ],
  [["a", "e", "i", "o", "u"], []],
  [
    [4, "z", "f", 5],
    [4, 5]
  ],
  [["abc", 123], [123]],
  [["$%^", 567, "&&&"], [567]],
  [
    ["w", "r", "u", 43, "s", "a", 76, "d", 88],
    [43, 76, 88]
  ]
];

describe("filterArray", () => {
  describe("Return new array without strings", () => {
    test.each(data)("filterArray(%p) = %p", (arr, result) => {
      let actual = filterArray(arr);

      expect(actual).toEqual(result);
    });
  });
});

File: CSharp/Tests/FilterStringsFromArrayTest.cs
using System.Collections.Generic;
using Xunit;

namespace CSharp.Tests
{
    public class FilterStringsFromArrayTest
    {
        [Theory]
        [MemberData(nameof(TestObjectArrayData))]
        public void FilterArray_ObjectArrayInput_ReturnIntArray(object[] arr, int[] expected)
        {
            var actual = FilterStringsFromArray.FilterArray(arr);

            Assert.Equal(expected, actual);
        }

        public static IEnumerable<object[]> TestObjectArrayData
        {
            get
            {
                return new[]
                {
                    new object[] { new object[] { 1, 2, "a", "b" }, new int[] { 1, 2 } },
                    new object[] { new object[] { 1, "a", "b", 0, 15 }, new int[] { 1, 0, 15 } },
                    new object[] { new object[] { 1, 2, "aasf", "1", "123", 123 }, new int[] { 1, 2, 123 } },
                    new object[] { new object[] { "jsyt", 4, "yt", 6 }, new int[] { 4,6 } },
                    new object[] { new object[] { "r", 5, "y", "e", 8, 9 }, new int[] { 5, 8, 9 } },
                    new object[] { new object[] { "a", "e", "i", "o", "u" }, new int[] {  } },
                    new object[] { new object[] { 4, "z", "f", 5 }, new int[] { 4, 5 } },
                    new object[] { new object[] { "abc", 123 }, new int[] { 123 } },
                    new object[] { new object[] { "$%^", 567, "&&&" }, new int[] { 567 } },
                    new object[] { new object[] { "w", "r", "u", 43, "s", "a", 76, "d", 88 }, new int[] { 43, 76, 88 } }
                };
            }
        }
    }
}